R/lm_robust_fit.R

```diff
@@ -65,58 +67,112 @@ lm_robust_fit <- function(y,
     # which_ests <- return_frame$variable_names %in% deparse(substitute(coefficient_name))
   }
 
-  if (!is.null(weights)) {
-    Xunweighted <- X
-    weight_mean <- mean(weights)
-    weights <- sqrt(weights / weight_mean)
-    X <- weights * X
-    y <- weights * y
+
+  if(ei) {
+
+    if (!is.null(cluster)) {
+      cl_ord <- order(cluster)
+      y <- y[cl_ord]
+      X <- X[cl_ord,]
+      cluster <- cluster[cl_ord]
+      J <- length(unique(cluster))
+      if (!is.null(weights)) {
+        weights <- weights[cl_ord]
+      }
+
+    } else {
+      J <- 1
+    }
+
```

Comment:
Here is where I reorder everything so that I only have to pass through the data once in RcppEigen.